python/ray/data/impl/arrow_block.py

```diff
@@ -399,3 +397,79 @@ def _copy_table(table: "pyarrow.Table") -> "pyarrow.Table":
             arr = col.combine_chunks()
         new_cols.append(arr)
     return pa.Table.from_arrays(new_cols, schema=table.schema)
+
+
+def _arrow_array_reduce(a: "pyarrow.Array"):
+    """Custom reducer for Arrow arrays that works around a zero-copy slicing pickling
+    bug.
+
+    See https://issues.apache.org/jira/browse/ARROW-10739.
+    """
+    # TODO(Clark): Do we need to generically support children arrays?
```

Comment:
I think this would be required for full support of nested array types (list arrays, struct arrays, map arrays, union arrays), but this may be difficult since the child arrays aren't easily accessible AFAICT.

Reply:
What does this mean if a parquet file is using structs? This is fairly common.

Reply:
To be clear, I think that this is a blocker for this getting merged. I'll add it to the PR description as a merge blocker.